src/test/java/uk/nhs/prm/e2etests/performance/RepositoryPerformanceTest.java

```diff
@@ -0,0 +1,124 @@
+    private final EhrTransferServiceUnhandledOQ ehrTransferServiceUnhandledOQ;
+    private final EhrTransferServiceLargeEhrFragmentsOQ ehrTransferServiceLargeEhrFragmentsOQ;
+    private final EhrTransferServiceSmallEhrOQ ehrTransferServiceSmallEhrOQ;
+    private final EhrTransferServiceLargeEhrOQ ehrTransferServiceLargeEhrOQ;
+    private final EhrTransferServiceNegativeAcknowledgementOQ ehrTransferServiceNegativeAcknowledgementOQ;
+    private final EhrTransferServiceParsingDeadLetterQueue ehrTransferServiceParsingDeadLetterQueue;
+
+    @Autowired
+    public RepositoryPerformanceTest(
+            RepoService repoService,
+            TemplatingService templatingService,
+            SimpleAmqpQueue mhsInboundQueue,
+            Gp2GpMessengerOQ gp2gpMessengerOQ,
+            EhrTransferServiceTransferCompleteOQ ehrTransferServiceTransferCompleteOQ,
+            EhrTransferServiceUnhandledOQ ehrTransferServiceUnhandledOQ,
+            EhrTransferServiceLargeEhrFragmentsOQ ehrTransferServiceLargeEhrFragmentsOQ,
+            EhrTransferServiceSmallEhrOQ ehrTransferServiceSmallEhrOQ,
+            EhrTransferServiceLargeEhrOQ ehrTransferServiceLargeEhrOQ,
+            EhrTransferServiceNegativeAcknowledgementOQ ehrTransferServiceNegativeAcknowledgementOQ,
+            EhrTransferServiceParsingDeadLetterQueue ehrTransferServiceParsingDeadLetterQueue
+    ) {
+        this.repoService = repoService;
+        this.templatingService = templatingService;
+        this.mhsInboundQueue = mhsInboundQueue;
+        this.gp2gpMessengerOQ = gp2gpMessengerOQ;
+        this.ehrTransferServiceTransferCompleteOQ = ehrTransferServiceTransferCompleteOQ;
+        this.ehrTransferServiceUnhandledOQ = ehrTransferServiceUnhandledOQ;
+        this.ehrTransferServiceLargeEhrFragmentsOQ = ehrTransferServiceLargeEhrFragmentsOQ;
+        this.ehrTransferServiceSmallEhrOQ = ehrTransferServiceSmallEhrOQ;
+        this.ehrTransferServiceLargeEhrOQ = ehrTransferServiceLargeEhrOQ;
+        this.ehrTransferServiceNegativeAcknowledgementOQ = ehrTransferServiceNegativeAcknowledgementOQ;
+        this.ehrTransferServiceParsingDeadLetterQueue = ehrTransferServiceParsingDeadLetterQueue;
+    }
+
+    @BeforeAll
+    void init() {
+        ehrTransferServiceSmallEhrOQ.deleteAllMessages();
+        ehrTransferServiceLargeEhrOQ.deleteAllMessages();
+        ehrTransferServiceLargeEhrFragmentsOQ.deleteAllMessages();
+        ehrTransferServiceParsingDeadLetterQueue.deleteAllMessages();
+        ehrTransferServiceTransferCompleteOQ.deleteAllMessages();
+        ehrTransferServiceUnhandledOQ.deleteAllMessages();
+        ehrTransferServiceNegativeAcknowledgementOQ.deleteAllMessages();
+        gp2gpMessengerOQ.deleteAllMessages();
+    }
+
+    @Test
+    void Given_SuperLargeEhrWith100Fragments_When_PutIntoRepoAndPulledOut_Then_VisibleOnGp2gpMessengerOQ() {
+        // given
+        String patientNhsNumber = "9727018157";
```

Comment:
I reckon renaming to `nhsNumber` would be better, as it's a given that it's a patient's.